File: frontend/src/app/core/services/clinical-support.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../../models/auth.model';
import { SafetyCheckRequest, SafetyAlert, ClinicalBrief } from '../../models/clinical-support.model';

@Injectable({
  providedIn: 'root'
})
export class ClinicalSupportService {
  private apiUrl = `${environment.apiUrl}/clinical-support`;

  constructor(private http: HttpClient) {}

  checkSafety(request: SafetyCheckRequest): Observable<ApiResponse<SafetyAlert[]>> {
    return this.http.post<ApiResponse<SafetyAlert[]>>(`${this.apiUrl}/check-safety`, request);
  }

  getPatientBrief(patientId: number): Observable<ApiResponse<ClinicalBrief>> {
    return this.http.get<ApiResponse<ClinicalBrief>>(`${this.apiUrl}/patient-brief/${patientId}`);
  }
}

File: backend/src/main/java/com/hospital/hms/service/ClinicalSupportService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.ClinicalSupportDTO;
import java.util.List;

public interface ClinicalSupportService {
    List<ClinicalSupportDTO.SafetyAlert> checkDrugSafety(ClinicalSupportDTO.SafetyCheckRequest request);
    ClinicalSupportDTO.ClinicalBrief generatePatientBrief(Long patientId);
}
